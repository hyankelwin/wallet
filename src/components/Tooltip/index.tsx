import React from 'react';
import { Container } from './styles';

interface TooltipProps {
  title: string;
  className?: string;
  small?: boolean;
}

const Tooltip: React.FC<TooltipProps> = ({
  title,
  children,
  className = '',
  small,
}) => {
  return (
    <Container className={className} small={small}>
      {children}
      <span>{title}</span>
    </Container>
  );
};

export default Tooltip;
