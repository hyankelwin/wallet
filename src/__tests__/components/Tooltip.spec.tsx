import { mount } from 'enzyme';
import Tooltip from '../../components/Tooltip';

describe('Tooltip component', () => {
  it('should render Tooltip component with success', () => {
    const wrapper = mount(<Tooltip title="teste" className="name-teste" />);

    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render Tooltip component not passed property className with success', () => {
    const wrapper = mount(<Tooltip title="teste" />);

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
