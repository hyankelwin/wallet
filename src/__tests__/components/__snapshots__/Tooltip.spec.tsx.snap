// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component should render Tooltip component not passed property className with success 1`] = `
"<Tooltip title=\\"teste\\">
  <styled.div className=\\"\\" small={[undefined]}>
    <div className=\\"sc-bdvvtL lbHHvh\\">
      <span>
        teste
      </span>
    </div>
  </styled.div>
</Tooltip>"
`;

exports[`Tooltip component should render Tooltip component with success 1`] = `
"<Tooltip title=\\"teste\\" className=\\"name-teste\\">
  <styled.div className=\\"name-teste\\" small={[undefined]}>
    <div className=\\"sc-bdvvtL lbHHvh name-teste\\">
      <span>
        teste
      </span>
    </div>
  </styled.div>
</Tooltip>"
`;
